polar-core/src/bindings.rs

```diff
@@ -16,9 +16,15 @@ pub struct Binding(pub Symbol, pub Term);
 pub type BindingStack = Vec<Binding>;
 pub type Bindings = HashMap<Symbol, Term>;
 
-pub type Bsp = usize;
+pub type Bsp = Bsps;
 pub type FollowerId = usize;
 
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Bsps {
+    bsp: usize,
```

Comment:
What do you think about renaming this `pointer` or `ptr` or something to indicate that this is _actually_ the bsp? Or maybe renaming the struct to something like `BspWithFollowers` or `BspTree`?

Currently finding things like `self.bsp().bsp` and `let bsp = bsp.bsp;` a little hard to read.

Reply:
Renamed to `bindings_index`.